cpp/include/ray/api/ray_config.h

```diff
@@ -9,45 +9,31 @@
 namespace ray {
 namespace api {
 
-enum class RunMode { SINGLE_PROCESS, CLUSTER };
-
-/// TODO(Guyang Song): Make configuration complete and use to initialize.
 class RayConfig {
  public:
-  WorkerType worker_type = WorkerType::DRIVER;
-
-  RunMode run_mode = RunMode::SINGLE_PROCESS;
-
-  std::string redis_ip;
-
-  int redis_port = 6379;
-
-  std::string redis_password = "5241590000000000";
+  // The redis address of existing cluster.
+  std::string address;
 
-  int node_manager_port = 62665;
+  // Whether or not to run this program in a single process. This is used for debugging.
+  bool local_mode = true;
 
-  std::string lib_name = "";
+  // The dynamic library path which contains remote fuctions of users.
+  // Local mode does not need to set this parameter.
+  std::string dynamic_library_path = "";
 
-  std::string store_socket = "";
+  /* The following are unstable parameters and their use is discouraged. */
 
-  std::string raylet_socket = "";
+  // Prevent external clients without the password from connecting.
+  std::string redis_password_ = "5241590000000000";
```

Comment:
We shouldn't write the default password here. Use `boost::optional<std::string>` instead.

Reply:
Maybe we will remove boost from headers later?